### ISSUE TITLE
terragrunt.hcl files linting

### ISSUE THREAD

**Issue by alikhil** [gruntwork-io/terragrunt#2807]
tflint now supports only linting terraform files. However, terragrunt is getting popular and widely used now (even inside a single company/project) and this requires to standardize terragrunt.hcl files. It would be nice to reuse tflint but now it does not support linting hcl files.

What is your vision about linting terragrunt configurations?

**Comment by denis256** [comment]
Hello,
existing solution with `tflint` hook is not useful?

References:
https://terragrunt.gruntwork.io/docs/features/hooks/#tflint-hook

**Comment by muzakh** [comment]
Please try command `terragrunt hclfmt`. hclfmt is a command to formatting and prettifying the HCL files.

**Comment by alikhil** [comment]
@wyardley You are right, I am asking about linting hcl configs for terragrunt itself. 

For example, I may want to deprecate including (`include` directive) some `provider.hcl` file since we are moving to `anotherprovider.hcl` and this could not be validated neither by tflint nor by hclfmt

**Comment by wyardley** [comment]
Commenting for activity

**Comment by yhakbar** [comment]
Hey folks,

This is a good request, and in line with other requests for Dev tooling we've heard from the community for things like an official LSP.

Unfortunately, we don't have the bandwidth to invest in tooling like this right now, though we acknowledge it would be powerful for the community.

We've recently added a bit more validation logic via `hclvalidate`, but it's not something that's going to catch bad practices or anything.

If anyone in the community is interested in starting a project like this, we'd appreciate it. We'd call it out in our docs if it was good quality, and help with testing it.

**Comment by yhakbar** [comment]
Terragrunt currently has the [`hcl validate` command](https://terragrunt.gruntwork.io/docs/reference/cli/commands/hcl/validate/) which will validate Terragrunt HCL configurations. It includes the [`--inputs`](https://terragrunt.gruntwork.io/docs/reference/cli/commands/hcl/validate/#inputs) and [`--strict`](https://terragrunt.gruntwork.io/docs/reference/cli/commands/hcl/validate/#strict) flags for detecting potential issues.

If there are patterns that would be useful to detect for better Terragrunt practices, we can consider enhancements to the `hcl validate` command.

We can also look to integrate this with the [Terragrunt Language Server](https://github.com/gruntwork-io/terragrunt-ls) once we've released 1.0, and can dedicate some time to stabilizing the language server.

Closing out this issue in favor of more specific enhancement requests.
